kyo-core/js/src/main/scala/kyo/KyoAppPlatformSpecific.scala

```diff
@@ -0,0 +1,41 @@
+package kyo
+
+import kyo.Maybe.Absent
+import kyo.Maybe.Present
+
+abstract class KyoAppPlatformSpecific extends KyoApp.Base[Async & Resource & Abort[Throwable]]:
+
+    private var previousFiberIO: Maybe[Fiber[Throwable, Unit] < IO] = Absent
+
+    final override protected def run[A: Flat](v: => A < (Async & Resource & Abort[Throwable]))(using Frame): Unit =
+        import AllowUnsafe.embrace.danger
+        val currentFiberIO =
+            for
+                currentFiber <- handle(v).flatMap(_.mapResult { result =>
+                    IO(Result(if !result.exists(().equals(_)) then println(pprint.apply(result).plainText)))
+                })
+                sleepFiber <- Async.run(Async.sleep(timeout))
+                raceFiber  <- Fiber.race(Seq(currentFiber.get, sleepFiber.get))
+            yield raceFiber
```

Comment:
I try to get the same behavior on the JVM side, by racing this with a `Async.sleep`